uoftscrapers/scrapers/events/__init__.py

```diff
@@ -0,0 +1,115 @@
+        event_url = Events.host + url_tail
+        html = Scraper.get(event_url)
+        url_parts = list(urlparse.urlparse(event_url))
+        query = dict(urlparse.parse_qsl(url_parts[4]))
+        soup = BeautifulSoup(html, 'html.parser')
+
+        event_id = query['eventid']
+        event_title = soup.select('.eventTitle')[0].text.strip()
+        raw_time = soup.select('.date')[0].text.split(',')
+        
+        date_arr = raw_time[0].split(' - ')
+        time_arr = re.split(' - | ', raw_time[1].strip())
+        
+        # Some of the strings are misformed and gives an extra empty space
+        time_arr = list(filter(None, time_arr))
+        event_start_date = datetime.strptime(date_arr[0],
+            '%b %d').replace(year=date.today().year).date().isoformat()
+        event_end_date = datetime.strptime(date_arr[-1], 
+            '%b %d').replace(year=date.today().year).date().isoformat()
+
+        # Note: Some events span across several days e.g. 8350, thus specifying dates makes no sense
+        event_meridiem = time_arr[2]
+        event_start_time = time_arr[0] + ' ' + event_meridiem
+        event_end_time = time_arr[1] + ' ' + event_meridiem
+        
+        evt_bar = soup.select('#evt_bar')[0]
+        event_url = evt_bar.select('dd')[1].a['href']
+        event_price = evt_bar.select('dl')[1].dd.text
+
+        event_campus = 'Off Campus'
+        if evt_bar.select('dd')[0].b != None:
+            event_campus = evt_bar.select('dd')[0].b.text
+
+        event_address = ''
+        address_block = evt_bar.select('dd')[0]
+        if address_block.a != None:
+            address_block = address_block.a
+        for content in address_block.contents:
+            text = content if type(content) == NavigableString else content.text
+            event_address += text.strip().replace('\r', '') + ' '
+        event_address = event_address.strip()
+
+        event_audiences = list(map(lambda a: a.text, 
+            evt_bar.select('dl')[1].select('dd')[1].select('a')))
+
+        soup.select('.eventTitle')[0].extract()
+        soup.select('.date')[0].extract()
+        evt_bar.extract()
+        soup.select('#cal_bar')[0].extract()
+        event_description = ''
+        for content in soup.select('#content')[0].contents:
+            text = content if type(content) == NavigableString else content.text
+            event_description += text.strip().replace('\r', '') + ' '
+        event_description = event_description.strip()
+
+        doc = OrderedDict([
+            ('id', event_id),
+            ('title', event_title),
+            ('start_date', event_start_date),
+            ('end_date', event_end_date),
+            ('start_time', event_start_time),
+            ('end_time', event_end_time),
+            ('url', event_url),
+            ('description', event_description),
+            ('admission_price', event_price),
+            ('campus', event_campus),
+            ('address', event_address),
```

Comment:
Should we rename this to `location` instead? It seems like the `event_address` often has values describing location instead of a well formatted physical address, which could be misleading.